modules/server/src/main/scala/io/chrisdavenport/ember/server/EmberServerBuilder.scala

```diff
@@ -0,0 +1,128 @@
+package io.chrisdavenport.ember.server
+
+import cats._
+import cats.implicits._
+import cats.effect._
+import fs2.concurrent._
+import org.http4s._
+import org.http4s.server.Server
+import scala.concurrent.duration._
+import java.net.InetSocketAddress
+import java.util.concurrent.Executors
+import java.nio.channels.AsynchronousChannelGroup
+
+
+final class EmberServerBuilder[F[_]: Concurrent: Timer: ContextShift] private (
+  private val host: String,
+  private val port: Int,
+  private val httpApp: HttpApp[F],
+  private val agR: Resource[F, AsynchronousChannelGroup],
+  private val onError: Throwable => Response[F],
+  private val onWriteFailure : Option[(Option[Request[F]], Response[F], Throwable) => F[Unit]],
+  private val maxConcurrency: Int,
+  private val receiveBufferSize: Int,
+  private val maxHeaderSize: Int,
+  private val requestHeaderReceiveTimeout: Duration
+){ self => 
+
+  private def copy(
+    host: String = self.host,
+    port: Int = self.port,
+    httpApp: HttpApp[F] = self.httpApp,
+    agR: Resource[F, AsynchronousChannelGroup] = self.agR,
+    onError: Throwable => Response[F] = self.onError,
+    onWriteFailure : Option[(Option[Request[F]], Response[F], Throwable) => F[Unit]] = self.onWriteFailure,
+    maxConcurrency: Int = self.maxConcurrency,
+    receiveBufferSize: Int = self.receiveBufferSize,
+    maxHeaderSize: Int = self.maxHeaderSize,
+    requestHeaderReceiveTimeout: Duration = self.requestHeaderReceiveTimeout
+  ): EmberServerBuilder[F] = new EmberServerBuilder[F](
+    host = host,
+    port = port,
+    httpApp = httpApp,
+    agR = agR,
+    onError = onError,
+    onWriteFailure = onWriteFailure,
+    maxConcurrency = maxConcurrency,
+    receiveBufferSize = receiveBufferSize,
+    maxHeaderSize = maxHeaderSize,
+    requestHeaderReceiveTimeout = requestHeaderReceiveTimeout
+  )
+
+  def withHost(host: String) = copy(host = host)
+  def withPort(port: Int) = copy(port = port)
+  def withHttpApp(httpApp: HttpApp[F]) = copy(httpApp = httpApp)
+  def withAsynchronousChannelGroup(acg: AsynchronousChannelGroup) =
+    copy(agR = acg.pure[Resource[F, ?]])
+  def withOnError(onError: Throwable => Response[F]) = copy(onError = onError)
+  def withOnWriteFailure(onWriteFailure: Option[(Option[Request[F]], Response[F], Throwable) => F[Unit]]) =
```

Comment:
Our other builders that take an `Option` usually call the field `Option`, and then have a `withOnWriteFailure` and `withoutWriteFailure`.  Those names are awkward, which I think you can work around without the outer `Option`.

Reply:
Dropped Option so we should be cleaner now.